legion_test/bin/cleanup_nexus_repo

```diff
@@ -102,119 +99,128 @@ def filter_builds_by_date(builds, remaining_days):
     remaining_date = datetime.datetime.now() - datetime.timedelta(remaining_days)
     builds_to_del = []
     for build in builds:
-        print(build)
         try:
-            build_date = re.search('\d{14}', build.get("version")).group()
-            if datetime.datetime.strptime(build_date, '%Y%m%d%H%M%S') < remaining_date:
+            build_date = [
+                re.search(VERSION_SEARCH_REGEX, build.get(field) if build.get(field) else '')
+                for field in FIELDS_TO_ANALYZE
+            ]
+            if not any(build_date):
+                log.warning('Can\'t find build date for {!r}'.format(build))
+                continue
+
+            regex_date = next(regex.group() for regex in build_date if regex)
+
+            if len(regex_date) == 12:
+                regex_pattern = '%y%m%d%H%M%S'
+            elif len(regex_date) == 14:
+                regex_pattern = '%Y%m%d%H%M%S'
+            else:
+                log.warning('Unknown length of matched date {!r}'.format(regex_date))
```

Comment:
If this branch executes than the regex_pattern variable will be undefined.

Reply:
done